iceoryx_examples/callbacks/ice_callbacks_subscriber.cpp

```diff
@@ -54,24 +81,39 @@ int main()
 
     iox::runtime::PoshRuntime::initRuntime("iox-ex-callbacks-subscriber");
 
+    // the listener starts a background thread and the callbacks of the attached events
+    // will be called in this background thread when they are triggered
     iox::popo::Listener listener;
 
-    // attach shutdownTrigger to handle CTRL+C
-    listener.attachEvent(shutdownTrigger, shutdownTriggerCallback);
-
-    iox::popo::Subscriber<CounterTopic> subscriber({"Radar", "FrontLeft", "Counter"});
-
-    subscriber.subscribe();
-
-    listener.attachEvent(subscriber, iox::popo::SubscriberEvent::HAS_DATA, onSampleReceived);
-
-    while (keepRunning)
-    {
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    }
-
-    listener.detachEvent(shutdownTrigger);
-    listener.detachEvent(subscriber, iox::popo::SubscriberEvent::HAS_DATA);
+    iox::popo::UserTrigger heartbeat;
+    iox::popo::Subscriber<CounterTopic> subscriberLeft({"Radar", "FrontLeft", "Counter"});
+    iox::popo::Subscriber<CounterTopic> subscriberRight({"Radar", "FrontRight", "Counter"});
+
+    // send a heartbeat every 4 seconds
+    std::thread heartbeatThread([&] {
+        while (keepRunning)
+        {
+            heartbeat.trigger();
+            std::this_thread::sleep_for(std::chrono::seconds(4));
+        }
+    });
+
+    // attach everything to the listener, from here one the callbacks are called when an event occurs
+    listener.attachEvent(heartbeat, heartbeatCallback);
+    listener.attachEvent(subscriberLeft, iox::popo::SubscriberEvent::HAS_DATA, onSampleReceivedCallback);
+    listener.attachEvent(subscriberRight, iox::popo::SubscriberEvent::HAS_DATA, onSampleReceivedCallback);
+
+    // wait until someone presses CTRL+c
+    mainLoopBlocker.wait();
```

Comment:
There is no main loop. This is just waiting for the termination signal for the main method, I suggest calling it differently to avoid confusion (shutdownSemaphore or something like this).